vpr/src/place/net_cost_handler.cpp

```diff
@@ -271,24 +282,24 @@ void NetCostHandler::alloc_and_load_for_fast_vertical_cost_update_(float place_c
         }
```

Comment:
Can't two inner loops be replaced with a single loop that iterates over all edges of the node?

Reply:
change the loops to the following format:

```
for (const auto& src_rr_node : rr_graph.nodes()) {
        for (auto edge_range: {rr_graph.configurable_edges(src_rr_node), rr_graph.non_configurable_edges(src_rr_node)}) {
            for (const auto& rr_edge_idx : edge_range) {
                const auto& sink_rr_node = rr_graph.edge_sink_node(src_rr_node, rr_edge_idx);
                if (rr_graph.node_layer(src_rr_node) != rr_graph.node_layer(sink_rr_node)) {
                    // We assume that the nodes driving the inter-layer connection or being driven by it
                    // are not stretched across multiple tiles
                    int src_x = rr_graph.node_xhigh(src_rr_node);
                    int src_y = rr_graph.node_yhigh(src_rr_node);
                    VTR_ASSERT(rr_graph.node_xlow(src_rr_node) == src_x && rr_graph.node_ylow(src_rr_node) == src_y);

                    tile_num_inter_die_conn[src_x][src_y]++;
                }
            }
        }
    }
```

Reply:
I am not sure about this, but isn't `edges()` the union of `configurable_edges` and `non_configurable_edges`?

Reply:
You are right. For some reason, I missed this function. Thanks!